### PR TITLE
Column title metrics

### PR DESCRIPTION
**Pull request by Crouchsnap**
## Overview
Adds a metric to count the number of times column titles are changed

### Notes
I refactored the repository calls in ColumnTitleController into a separate service and made the metrics call there.

### CODE REVIEW

File: api/src/main/java/com/ford/labs/retroquest/columntitle/ColumnTitleService.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    public ColumnTitle editColumnTitleName(Long columnId, String newColumnName, String teamId) {
+        var existingColumnTitle = columnTitleRepository.findById(columnId).orElseThrow(ColumnTitleNotFoundException::new);
+        existingColumnTitle.setTitle(newColumnName);
+
+        ColumnTitle newColumnTitle = columnTitleRepository.save(existingColumnTitle);
+
+        websocketService.publishEvent(new WebsocketColumnTitleEvent(teamId, WebsocketEventType.UPDATE, newColumnTitle));
+
+        meterRegistry.counter("retroquest.columns.changed.count").increment();
```
Comment (LowBudgetMan) [inline api/src/main/java/com/ford/labs/retroquest/columntitle/ColumnTitleService.java:37]: I know there's an `@Timer` annotation that can be added to a method to get both count and timing metrics for a function. I wonder if there's also one for the counter? Unless we specifically only want to count the number of successful change requests and not the number of change requests total.

**Review by LowBudgetMan — CHANGES_REQUESTED** [review]

File: api/src/test/java/com/ford/labs/retroquest/columntitle/ColumnTitleServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        String newColumnName = "Some new Title";
+        var savedColumnTitle = columnTitleService.editColumnTitleName(columnTitle.getId(), newColumnName, teamId );
+
+        assertThat(meterRegistry.get("retroquest.columns.changed.count").counter().count()).isEqualTo(1);
+        assertThat(savedColumnTitle.getTitle()).isEqualTo(newColumnName);
+        assertThat(savedColumnTitle.getId()).isEqualTo(columnTitle.getId());
+    }
+
+    @Test
+    void throws_column_title_not_found_exception_when_column_title_not_in_db() {
```
[the remark this answers is not in the document]
Comment (Crouchsnap) [reply api/src/test/java/com/ford/labs/retroquest/columntitle/ColumnTitleServiceTest.java:63]: Uhh, I guess you could? but it returns a list, so it would just send an empty list back. It's kind of a weird edge case, because column titles are generated when you make a team; so it would presumably be caused by trying to look up the titles for a team that doesn't exist.

File: api/src/main/java/com/ford/labs/retroquest/columntitle/ColumnTitleService.java (hunk 1 above)
Comment (Crouchsnap) [reply api/src/main/java/com/ford/labs/retroquest/columntitle/ColumnTitleService.java:37]: I guess we could count every attempt to change a column title? We're just looking to understand whether folks change the titles more often or not after we publish some information about retrospectives, not add instrumentation looking for failure cases or anything.

Comment (LowBudgetMan) [reply api/src/main/java/com/ford/labs/retroquest/columntitle/ColumnTitleService.java:37]: Alright, this looks good enough for now then.

File: api/src/test/java/com/ford/labs/retroquest/columntitle/ColumnTitleServiceTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (LowBudgetMan) [reply api/src/test/java/com/ford/labs/retroquest/columntitle/ColumnTitleServiceTest.java:63]: I'll make a card for this then. There are other spots in the app where this is a problem too.

**Review by LowBudgetMan — APPROVED** [review]
:shipit:
